src/web/core/src/main/java/org/geoserver/web/wicket/browser/FileRootsFinder.java

```diff
@@ -0,0 +1,176 @@
+
+        String separator;
+        boolean dataDirectoryPath;
+        String base;
+        String name;
+        
+        public PathSplitter(String input, boolean dataDirectoryPath) {
+            // decide which separator to use based on data dir vs actual file system
+            this.separator = dataDirectoryPath ? "/" : File.separator;
+            this.dataDirectoryPath = dataDirectoryPath;
+            
+            // remove protocol part if needed (
+            if (input.startsWith("file:")) {
+                if (input.startsWith("file:/")) {
+                    input = input.substring(6);
+                } else {
+                    input = input.substring(5);
+                }
+            }
+
+            // split input into base and last segment
+            int idx = input.lastIndexOf(separator);
+            if (idx >= 0) {
+                base = input.substring(0, idx);
+                name = input.substring(idx + 1, input.length()).toLowerCase();
+            } else {
+                base = separator;
+                name = input.toLowerCase();
+            }
+
+            // fix base in case of data dir
+            if (dataDirectoryPath) {
+               base = Paths.convert(base); 
+            }
+        }
+        
+        private String buildPath(String name) {
+            // Data dir relative path weirdness, the protocol has to be 
+            // file:/ instead of file:// or it won't work. 
+            String prefix = dataDirectoryPath ? "file:" : "file://";
+            // make data dir relative paths actually relative despite user's input
+            String localBase = base;
+            if(dataDirectoryPath && localBase.startsWith(separator)) {
+                localBase = base.substring(1);
+            }
+            if (localBase.endsWith(separator)) {
+                return prefix + localBase + name;
+            } else {
+                return prefix + localBase + separator + name;
+            }
+        }
+
+    }
+
+    private ArrayList<File> roots;
+    private File dataDirectory;
+
+    public FileRootsFinder(boolean includeDataDir) {
+        this(GeoServerFileChooser.HIDE_FS, includeDataDir);
+    }
+    
+    public FileRootsFinder(boolean hideFileSystem, boolean includeDataDir) {
+        // build the roots
+        roots = new ArrayList<File>();
+        if (!hideFileSystem) {
+            roots.addAll(Arrays.asList(File.listRoots()));
+        }
+        Collections.sort(roots);
+
+        // TODO: find a better way to deal with the data dir
+        GeoServerResourceLoader loader = getLoader();
+        dataDirectory = loader.getBaseDirectory();
+
+        if (includeDataDir) {
+            roots.add(0, dataDirectory);
+        }
+
+        // add the home directory as well if it was possible to determine it at all
+        if(!hideFileSystem && GeoServerFileChooser.USER_HOME != null) {
+            roots.add(1, GeoServerFileChooser.USER_HOME);
+        }
```

Comment:
I would *love* to have the https://github.com/geoserver/geoserver/wiki/GSIP-145 environmental variables added to this list of well-known paths to browse from. Not sure if it is possible to do though ...

Reply:
See above, have fun doing it yourself.

Reply:
For sure, I will ask on the email list if it is possible